diagnose_agent_banks.py

```
@@ -0,0 +1,134 @@
+# Check ports
+print("\n🔌 Port Status:")
+ports = {
+    5002: "CUA Execution Server (CRITICAL for real execution)",
+    7777: "Agent-Banks Web Interface",
+    3000: "Memory Service",
+    8888: "Claude's Realm IDE"
+}
+
+for port, service in ports.items():
+    try:
+        result = subprocess.run(['lsof', '-i', f':{port}'], capture_output=True, text=True)
```

Ruff (0.12.2)

`subprocess` call: check for execution of untrusted input

(S603)